### PR TITLE
initial commit of docker-registry stuff

### PR DESCRIPTION
**Pull request by Lorquas**

### CODE REVIEW

File: roles/docker/tasks/RedHat.yml [hunk 1]
```diff
@@ -5,3 +5,11 @@
...
   with_items:
     - docker-io
+
+- name: Ensure that Docker Registry and its dependencies are present
+  yum:
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - docker-registry
+  when: '"registry" in docker_roles'
```
Comment (ssalevan) [inline roles/docker/tasks/RedHat.yml]: Add a newline to this file real quick

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,9 @@
+*.pyc
```
Comment (ssalevan) [inline .gitignore:1]: +1, this is awesome

File: roles/docker/defaults/main.yml [hunk 3]
```diff
@@ -8,3 +8,8 @@ docker_authenticate: False
 docker_email: dev@null.com
 docker_username: username
 docker_password: password
+
```
Comment (ssalevan) [inline roles/docker/defaults/main.yml:11]: You'll want to add something to this file like:

docker_roles:
- docker

File: roles/docker/defaults/main.yml [hunk 4]
```diff
@@ -8,3 +8,8 @@ docker_authenticate: False
...
+docker_registry_port: 5000
+
+docker_failure_cycles_before_restart: 2
+docker_registry_failure_cycles_before_restart: 2
```
Comment (ssalevan) [inline roles/docker/defaults/main.yml]: Add a newline to the end of this file real quick

**Comment by ssalevan** [comment]
LGTM, just a few nits then should be mergeable

File: roles/docker/defaults/main.yml (hunk 3 above)
Comment (Lorquas) [inline roles/docker/defaults/main.yml:11]: you already added that in your last commit to the top of this very file.

**Comment by Lorquas** [comment]
newlines added and merge/squashed

**Pull request merged**